components/portal/main/services.js

```diff
@@ -64,20 +64,78 @@ define(['angular'], function(angular) {
+       * set the window title
+       *
+       * results in title
+       * page-title | app-title | portal-title (in an application), or
+       * page-title | portal-title (if the app has same name as the portal), or
+       * app-title | portal-title (if page name unknown or redundant)
+       * portal-title (if page name unknown or redundant and app name redundant)
+       *
+       * Examples:
+       * "Timesheets | STAR Time Entry | MyUW" ,
+       * for the Timesheets page in an app named "STAR Time Entry" in
+       * a portal named "MyUW", or
+       *
+       * "Notifications | MyUW",
+       * for the notifications page in an app named "MyUW" in
+       * a portal named "MyUW".
+       *
+       * "STAR Time Entry | MyUW"
+       * in an app named "STAR Time Entry" in a portal named "MyUW"
+       * when the page name is unspecified.
+       *
+       * "MyUW"
+       * in an app named "MyUW" in a portal named "MyUW"
+       * when the page name is unspecified.
        */
-      function setTitle() {
-        var frameTitle = '';
-        if ($rootScope.portal && $rootScope.portal.theme) {
-          frameTitle = $rootScope.portal.theme.title;
-          if (frameTitle !== NAMES.title && !APP_FLAGS.isWeb) {
-            frameTitle = ' | ' + frameTitle;
-          } else {
-            // since frame title equals the title in NAMES lets not duplicate it
-            frameTitle = '';
+      function setTitle(pageTitle) {
+        var windowTitle = ''; // we finally set the window title to this.
+
+        // first, gather pieces of the desired window title
+
+        var portalTitle = ''; // the name of the portal if we can determine it
+        var appTitle = ''; // the name of the app
+
+        if ($rootScope.portal && $rootScope.portal.theme &&
+              $rootScope.portal.theme.title) {
+          // there's an active theme with a title.
+          // consider that title the name of the portal
+          portalTitle = $rootScope.portal.theme.title;
+        }
+
+        appTitle = NAMES.title;
+
+        // assemble the window title from the gathered pieces,
+        // starting from the end of the window title and working back to the
+        // beginning.
+
+        // if the portal has a name, end the window title with that
+        // (if the portal lacks a name, portalTitle is still empty string)
+        windowTitle = portalTitle;
+
+        // if the app name differs from the portal, prepend it.
+        // if it's the same name, omit it to avoid silly redundancy like
+        // "MyUW | MyUW"
+        if (appTitle !== portalTitle) {
+          // if the windowTitle already has content, first add a separator
+          if (windowTitle !== '') {
+            windowTitle = ' | ' + windowTitle;
+          }
+          windowTitle = appTitle + windowTitle;
+        }
+
+        // if there's a page name not redundant with the app name, prepend it.
+        if (pageTitle && pageTitle !== '' && pageTitle !== appTitle) {
```

Comment:
`if (pageTitle && pageTitle !== appTitle) {` Should be enough, check the `If()` tab here: https://dorey.github.io/JavaScript-Equality-Table/